src/imperator/tables/modifs.rs

```diff
@@ -0,0 +1,378 @@
+    // local_$PopType$_desired_pop_ratio
+    if let Some(part) = name.as_str().strip_prefix("local_") {
+        for sfx in &["_output", "_desired_pop_ratio", "_happyness"] {
+            if let Some(part) = part.strip_suffix(sfx) {
+                if warn {
+                    data.verify_exists_implied(Item::PopType, part, name);
+                }
+                return Some(ModifKinds::Province | ModifKinds::State);
+            }
+        }
+    }
+
+    // global_$PopType$_output
+    // global_$PopType$_happyness
+    // global_$PopType$_desired_pop_ratio
+    // global_$PopType$_city_desired_pop_ratio
+    if let Some(part) = name.as_str().strip_prefix("global_") {
+        for sfx in &["_output", "_desired_pop_ratio", "_city_desired_pop_ratio", "_happyness"] {
+            if let Some(part) = part.strip_suffix(sfx) {
+                if warn {
+                    data.verify_exists_implied(Item::PopType, part, name);
+                }
+                return Some(ModifKinds::Country);
+            }
+        }
+    }
+
+    // $Price$_cost_modifier
+    if let Some(part) = name.as_str().strip_suffix("_cost_modifier") {
+        if warn {
+            data.verify_exists_implied(Item::Price, part, name);
+        }
+        return Some(ModifKinds::Country);
+    }
+
+    // $Party$_party_influence
+    if let Some(part) = name.as_str().strip_suffix("_party_influence") {
+        if warn {
+            data.verify_exists_implied(Item::PartyType, part, name);
+        }
+        return Some(ModifKinds::Country);
+    }
+
+    // monthly_$Party$_party_conviction
+    if let Some(part) = name.as_str().strip_prefix("monthly_") {
+        if let Some(part) = part.strip_suffix("_party_conviction") {
+            if warn {
+                data.verify_exists_implied(Item::PartyType, part, name);
+            }
+            return Some(ModifKinds::Country);
+        }
+    }
+
+    // $Unit$_discipline
+    // $Unit$_morale
+    // $Unit$_offensive
+    // $Unit$_defensive
+    // $Unit$_maintenance_cost
+    // $Unit$_movement_speed
+    for sfx in &["_discipline", "_morale", "_offensive", "_defensive", "_maintenance_cost", "_movement_speed"] {
+        if let Some(part) = name.as_str().strip_suffix(sfx) {
+            if warn {
+                data.verify_exists_implied(Item::Unit, part, name);
+            }
+            return Some(ModifKinds::Country);
+        }
+    }
+
+    // $Terrain$_combat_bonus
+    // $Unit$_$Terrain$_combat_bonus
+    if let Some(part) = name.as_str().strip_suffix("_combat_bonus") {
+        if warn
+            && !data.item_exists(Item::Terrain, part)
+            && !data.item_exists(Item::Unit, s)
+        {
+            let msg = format!("could not find any {part}");
+            let info = "Could be: $Terrain$_combat_bonus or $Unit$_$Terrain$_combat_bonus";
+            warn_info(name, ErrorKey::MissingItem, &msg, info);
+        }
+        return Some(ModifKinds::Country);
+    }
```

Comment:
This needs to be more complicated... for the `$Unit$_$Terrain$_combat_bonus` variant, you need to match both a unit and a terrain. Vic3 has one like that too, `building_group_$BuildingGroup$_$PopType$_fertility_mult`. You can look there for an example of how to handle this.

It'll be easier if neither unit nor terrain are likely to have `_` in their names.